scripts/coccinelle/null/badzero.cocci

```diff
@@ -0,0 +1,238 @@
+/// Compare pointer-typed values to NULL rather than 0
+///
+//# This makes an effort to choose between !x and x == NULL.  !x is used
+//# if it has previously been used with the function used to initialize x.
+//# This relies on type information.  More type information can be obtained
```

Comment:
can you change it to always do `x == NULL` ? We're going to follow MISRA-C guidelines and this is the recommended way to check pointers.